lib/src/chocolatey.dart

```diff
@@ -0,0 +1,279 @@
+String get _nupkgProperties {
+  var metadata = _nuspecMetadata;
+
+  var builder = XmlBuilder();
+  builder.processing("xml", 'version="1.0"');
+  builder.element("coreProperties", nest: () {
+    builder.namespace(
+        "http://schemas.openxmlformats.org/package/2006/metadata/core-properties");
+    builder.namespace("http://purl.org/dc/elements/1.1/", "dc");
+    builder.element("dc:creator", nest: _findElement(metadata, "authors").text);
+    builder.element("dc:description",
+        nest: _findElement(metadata, "description").text);
+    builder.element("dc:identifier", nest: _chocolateyName);
+    builder.element("version", nest: _chocolateyVersion);
+
+    var tags = _findElement(metadata, "tags", allowNone: true);
+    if (tags != null) builder.element("keywords", nest: tags.text);
+
+    var title = _findElement(metadata, "title", allowNone: true);
+    if (title != null) builder.element("dc:title", nest: title.text);
+  });
+  return builder.build().toString();
+}
+
+/// Whether [addChocolateyTasks] has been called yet.
+var _addedChocolateyTasks = false;
+
+/// Enables tasks for building and uploading packages to Chocolatey.
+void addChocolateyTasks() {
+  if (_addedChocolateyTasks) return;
+  _addedChocolateyTasks = true;
+
+  addStandaloneTasks();
+
+  // TODO(nweiz): Rather than publishing a snapshot, publish a script that
+  // downloads the tagged source and builds an executable at install-time. This
+  // will ensure that 64-bit Chocolatey users get native executable performance
+  // even if the package isn't built on Windows.
+  addTask(GrinderTask('pkg-chocolatey-build',
+      taskFunction: () => _build(),
+      description: 'Build a package to upload to Chocolatey.',
+      depends: ['pkg-compile-snapshot']));
+
+  addTask(GrinderTask('pkg-chocolatey-deploy',
+      taskFunction: () => _deploy(),
+      description: 'Deploy the Chocolatey package to Chocolatey.',
+      depends: ['pkg-chocolatey-build']));
+}
+
+/// Builds a package to upload to Chocolatey.
+Future<void> _build() async {
+  ensureBuild();
+
+  var archive = Archive()
+    ..addFile(fileFromString("$_chocolateyName.nuspec", _nuspec.toString()))
+    ..addFile(file("[Content_Types].xml",
+        p.join(await cliPkgSrc, "assets/chocolatey/[Content_Types].xml")))
+    ..addFile(fileFromString("_rels/.rels",
+        renderTemplate("chocolatey/rels.xml", {"name": _chocolateyName})))
+    ..addFile(fileFromString(
+        "package/services/metadata/core-properties/properties.psmdcp",
+        _nupkgProperties));
+
+  if (File("LICENSE").existsSync()) {
+    archive.addFile(file("tools/LICENSE", "LICENSE"));
+  }
+
+  for (var entrypoint in entrypoints) {
+    var snapshot = "${p.basename(entrypoint)}.snapshot";
+    archive.addFile(file("tools/$snapshot", "build/$snapshot"));
+  }
+
+  var install = StringBuffer();
+  var uninstall = StringBuffer();
+  executables.forEach((name, path) {
+    var args = '"$name" "\$packageFolder\\tools\\$name.bat"';
```

Comment:
This took me a while to parse, and I wasn't sure until I saw the expectations in the tests; I'd suggest adding a comment here about how the output will have a shell variable `$packageFolder`, or maybe a comment above `install` and `uninstall` that these StringBuffers will contain PowerShell script content?

Reply:
Done.